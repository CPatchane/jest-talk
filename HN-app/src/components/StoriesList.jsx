import React from 'react'

const StoriesList = (props) => {
  const { stories } = props
  return (
    <div className='stories'>
      {stories.map((s, i) => (
        <div key={i}>
          <a href={s.url} target='_blank'>
            <div>
              <span className='index'>{i + 1}</span>
              {`${s.title} by ${s.by}`}
              <span className='score'>
                {s.score} points
              </span>
            </div>
            <p className='link'>{s.url}</p>
          </a>
        </div>)
      )}
    </div>
  )
}

export default StoriesList

/*

V1 :

<a href={s.url} target='_blank'>
  {`${s.title} by ${s.by}`}
</a>

V2 :

<a href={s.url} target='_blank'>
  <div>
    <span className='index'>{i + 1}</span>
    {`${s.title} by ${s.by}`}
    <span className='score'>
      {s.score} points
    </span>
  </div>
  <p className='link'>{s.url}</p>
</a>

*/
