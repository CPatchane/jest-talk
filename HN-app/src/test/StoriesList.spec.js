/* eslint-env jest */

import React from 'react'

import StoriesList from '../../src/components/StoriesList'

import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
Enzyme.configure({ adapter: new Adapter() })

const mockStories = [
  {
    id: 1,
    title: 'Story number one',
    by: 'MissOne',
    score: 189,
    url: 'http://mock.example.com/mock1'
  },
  {
    id: 2,
    title: 'Story number two',
    by: 'MisterTwo',
    score: 145,
    url: 'http://mock.example.com/mock2'
  }
]

describe('StoriesList component', () => {
  it('should be rendered correctly', () => {
    const component = shallow(
      <StoriesList stories={mockStories} />
    ).getElement()
    expect(component).toMatchSnapshot()
  })
})
