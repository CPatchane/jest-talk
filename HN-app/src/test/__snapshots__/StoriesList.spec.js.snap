// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoriesList component should be rendered correctly 1`] = `
<div
  className="stories"
>
  <div>
    <a
      href="http://mock.example.com/mock1"
      target="_blank"
    >
      <div>
        <span
          className="index"
        >
          1
        </span>
        Story number one by MissOne
        <span
          className="score"
        >
          189
           points
        </span>
      </div>
      <p
        className="link"
      >
        http://mock.example.com/mock1
      </p>
    </a>
  </div>
  <div>
    <a
      href="http://mock.example.com/mock2"
      target="_blank"
    >
      <div>
        <span
          className="index"
        >
          2
        </span>
        Story number two by MisterTwo
        <span
          className="score"
        >
          145
           points
        </span>
      </div>
      <p
        className="link"
      >
        http://mock.example.com/mock2
      </p>
    </a>
  </div>
</div>
`;
